scripts/performance/perf_processes.py

```diff
@@ -323,7 +343,13 @@ class RGSchema(RequestGenerator):
 
     def _from_file_str_data(self, file_str):
         req_json = super()._from_file_str_data(file_str)
-        schema_id = json.loads(req_json)['result']['txnMetadata']['txnId']
+        if req_json is None:
+            return None
+        tmp = json.loads(req_json)
+        txn_type = (tmp.get('result', {}).get('txn', {}) or tmp.get('txn', {})).get('type', None)
+        if txn_type not in ["101", "107"]:
+            return None
+        schema_id = (tmp.get('result', {}).get('txnMetadata', {}) or tmp.get('txnMetadata', {})).get('txnId', None)
```

Comment:
This looks like duplicate code in multiple places. Can we use some utility/helper method?